src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Providers/Providers/IaasVmPsBackupProvider.cs

```diff
@@ -181,10 +181,52 @@ public List<AzureRmRecoveryServicesJobBase> ModifyPolicy()
             return jobsList;
         }
 
+        public List<AzureRmRecoveryServicesContainerBase> ListProtectionContainers()
+        {
+            string name = (string)this.ProviderData.ProviderParameters[ContainerParams.Name];
+            ContainerRegistrationStatus status = (ContainerRegistrationStatus)this.ProviderData.ProviderParameters[ContainerParams.Status];
+            ARSVault vault = (ARSVault)this.ProviderData.ProviderParameters[ContainerParams.Vault];
+            string resourceGroupName = (string)this.ProviderData.ProviderParameters[ContainerParams.ResourceGroupName];
+
+            ProtectionContainerListQueryParams queryParams = new ProtectionContainerListQueryParams();
+
+            // 1. Filter by Name
+            queryParams.FriendlyName = name;
+
+            // 2. Filter by ContainerType
+            queryParams.ProviderType = ProviderType.AzureIaasVM.ToString();
+
+            // 3. Filter by Status
+            queryParams.RegistrationStatus = status.ToString();
+
+            var listResponse = HydraAdapter.ListContainers(vault.ResouceGroupName, vault.Name, queryParams);
+
+            List<AzureRmRecoveryServicesContainerBase> containerModels = ConversionHelpers.GetContainerModelList(listResponse);
```

Comment:
above all lines are duplicate code for every provider. Pls check if list can be provider agnostic once again.